src/MultiContainerTrait.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace atk4\core;
+
+/**
+ * This trait makes it possible for you to add child objects
+ * into your object, but unlike "ContainerTrait" you can use
+ * multiple collections stored as different array properties.
+ *
+ * This class does not offer automatic naming, so if you try
+ * to add another element with same name, it will result in
+ * exception.
+ */
+trait MultiContainerTrait
+{
+    /**
+     * Use this method trait like this:.
+     *
+     * function addField($name, $definition) {
+     *     $field = $this->factory($definition, [], '\atk4\data\Field');
+     *
+     *     return $this->_addIntoCollection($name, $field, 'fields');
+     * }
+     *
+     * @param string $name       Name that can be used to reference object
+     * @param object $object     New element to add
+     * @param string $collection string String corresponding to the name of the property
+     *
+     * @throws Exception
+     *
+     * @return object|mixed $obect
+     */
+    public function _addIntoCollection(string $name, object $object, string $collection)
+    {
+        if (!$collection || !isset($this->{$collection}) || !is_array($this->{$collection})) {
```

Comment:
as argument is defined as string, can be better check for empty(!$collection)?

Reply:
And what about objects

Reply:
object LGTM => https://wiki.php.net/rfc/object-typehint

for empty @DarkSide666 you are right but i never used this way, i checked and is OK : 
http://sandbox.onlinephpfunctions.com/code/e9f41cac01daf4c17a583baa11f07e5503deebed